dtm0/fop.c

```diff
@@ -369,28 +400,36 @@ static int dtm0_fom_tick(struct m0_fom *fom)
 
 		if (m0_dtm0_in_ut() && req->dtr_msg == DMT_EXECUTE &&
 		    m0_dtm0_is_a_persistent_dtm(fom->fo_service)) {
-			rc = m0_dtm0_tx_desc_copy(&req->dtr_txr,
-						  &rep->dr_txr);
-			M0_ASSERT(rc == 0);
+			rc = m0_dtm0_tx_desc_copy(&req->dtr_txr, &rep->dr_txr);
+			if (rc != 0)
+				m0_fom_phase_move(fom, M0_ERR(rc), M0_FOPH_FAILURE);
 			rc = m0_dtm0_send_msg(fom, DMT_EXECUTED,
 					      &req->dtr_txr.dtd_id.dti_fid,
 					      &req->dtr_txr);
-			M0_ASSERT(rc == 0);
+			if (rc != 0)
+				m0_fom_phase_move(fom, M0_ERR(rc), M0_FOPH_FAILURE);
+		} else if (req->dtr_msg == DTM_PERSISTENT) {
+			if (m0_dtm0_is_a_volatile_dtm(fom->fo_service)) {
+				m0_be_dtm0_log_pmsg_post(svc->dos_log, fom->fo_fop);
+			} else {
+				m0_dtm0_logrec_update(svc->dos_log, &fom->fo_tx.tx_betx, &req->dtr_txr, &buf);
+				if (m0_dtm0_in_ut()) {
+					rc = m0_dtm0_tx_desc_copy(&req->dtr_txr, &rep->dr_txr);
+					if (rc != 0)
```

Comment:
I guess, if we are inside a UT then we can just use assertions here.

Reply:
OK.